ui/org.eclipse.pde.ui/src/org/eclipse/pde/internal/ui/editor/context/InputContextManager.java

```diff
@@ -208,30 +208,48 @@ public void resourceChanged(IResourceChangeEvent event) {
 		IResourceDelta delta = event.getDelta();
 
 		try {
+			List<IFile> added = new ArrayList<>();
+			List<IFile> removed = new ArrayList<>();
 			delta.accept(delta1 -> {
-				int kind = delta1.getKind();
 				IResource resource = delta1.getResource();
-				if (resource instanceof IFile) {
+				if (resource.getType() == IResource.FILE) {
+					int kind = delta1.getKind();
```

Comment:
Please filter files (match against actually monitored) before pushing everyone to UI thread (see this is currently done here: https://github.com/eclipse-pde/eclipse.pde/blob/fdbfcab2f6d2013bad76c187b5e47fcc6324ecd8/ui/org.eclipse.pde.ui/src/org/eclipse/pde/internal/ui/editor/context/InputContextManager.java#L256 )

With that we will only have ~5 files to be processed in UI, not 100_000.